README
NAME
    Dancer::Plugin::Email - Simple email sending for Dancer applications

VERSION
    version 1.0201

SYNOPSIS
        use Dancer;
        use Dancer::Plugin::Email;
    
        post '/contact' => sub {
            email {
                from    => 'bob@foo.com',
                to      => 'sue@foo.com',
                subject => 'allo',
                body    => 'Dear Sue, ...',
                attach  => '/path/to/attachment',
            };
        };

DESCRIPTION
    This plugin tries to make sending emails from Dancer applications as
    simple as possible. It uses Email::Sender under the hood. In a lot of
    cases, no configuration is required. For example, if your app is hosted
    on a unix-like server with sendmail installed, calling "email()" will
    just do the right thing.

    IMPORTANT: Version 1.x of this module is not backwards compatible with
    the 0.x versions. This module was originally built on Email::Stuff which
    was built on Email::Send which has been deprecated in favor of
    Email::Sender. Versions 1.x and on have be refactored to use
    Email::Sender. I have tried to keep the interface the same as much as
    possible. The main difference is the configuration. If there are
    features missing that you were using in older versions, then please let
    me know by creating an issue on github
    <https://github.com/ironcamel/Dancer-Plugin-Email>.

FUNCTIONS
    This module by default exports the single function "email".

  email
    This function sends an email. It takes a single argument, a hashref of
    parameters. Default values for the parameters may be provided in the
    headers section of the "CONFIGURATION". Paramaters provided to this
    function will override the corresponding configuration values if there
    is any overlap. An exception is thrown if sending the email fails, so
    wrapping calls to "email" with try/catch is recommended.

        use Dancer;
        use Dancer::Plugin::Email;
        use Try::Tiny;

        post '/contact' => sub {
            try {
                email {
                    sender  => 'bounces-here@foo.com', # optional
                    from    => 'bob@foo.com',
                    to      => 'sue@foo.com, jane@foo.com',
                    subject => 'allo',
                    body    => 'Dear Sue, ...<img src="cid:blabla">',
                    multipart => 'related', # optional, see below
                    attach  => [
                        '/path/to/attachment1',
                        '/path/to/attachment2',
                        {
                            Path => "/path/to/attachment3",
                            # Path is required when passing a hashref.
                            # See Mime::Entity for other optional values.
                            Id => "blabla",
                        }
                    ],
                    type    => 'html', # can be 'html' or 'plain'
                    # Optional extra headers
                    headers => {
                        "X-Mailer"          => 'This fine Dancer application',
                        "X-Accept-Language" => 'en',
                    }
                };
            } catch {
                error "Could not send email: $_";
            };
        };

CONFIGURATION
    No configuration is necessarily required. Email::Sender::Simple tries to
    make a good guess about how to send the message. It will usually try to
    use the sendmail program on unix-like systems and SMTP on Windows.
    However, you may explicitly configure a transport in your configuration.
    Only one transport may be configured. For documentation for the
    parameters of the transport, see the corresponding
    Email::Sender::Transport::* module. For example, the parameters
    available for the SMTP transport are documented here "ATTRIBUTES" in
    Email::Sender::Transport::SMTP.

    You may also provide default headers in the configuration:

        plugins:
          Email:
            # Set default headers (OPTIONAL)
            headers:
              sender: 'bounces-here@foo.com'
              from: 'bob@foo.com'
              subject: 'default subject'
              X-Mailer: 'MyDancer 1.0'
              X-Accept-Language: 'en'
            # Explicity set a transport (OPTIONAL)
            transport:
              Sendmail:
                sendmail: '/usr/sbin/sendmail'

    Example configuration for sending mail via Gmail:

        plugins:
          Email:
            transport:
              SMTP:
                ssl: 1
                host: 'smtp.gmail.com'
                port: 465
                sasl_username: 'bob@gmail.com'
                sasl_password: 'secret'

    Use the Sendmail transport using the sendmail program in the system
    path:

        plugins:
          Email:
            transport:
              Sendmail:

    Use the Sendmail transport with an explicit path to the sendmail
    program:

        plugins:
          Email:
            transport:
              Sendmail:
                sendmail: '/usr/sbin/sendmail'

  Multipart messages
    You can embed images in HTML messages this way: first, set the "type" to
    "html". Then pass the attachments as hashrefs, setting "Path" and "Id".
    In the HTML body, refer to the attachment using the "Id", prepending
    "cid:" in the "src" attribute. This works for popular webmail clients
    like Gmail and OE, but is not enough for Thunderbird, which wants a
    "multipart/related" mail, not the default "multipart/mixed". You can fix
    this adding the "multipart" parameter set to "related", which set the
    desired subtype when you pass attachments.

    Example:

      email {
             from    => $from,
             to      => $to,
             subject => $subject,
             body    => q{<p>Image embedded: <img src="cid:mycid"/></p>},
             type    => 'html',
             attach  => [ { Id => 'mycid', Path => '/path/to/file' }],
             multipart => 'related'
            };

CONTRIBUTORS
    *   Marco Pessotto <melmothx@gmail.com>

    *   Oleg A. Mamontov <oleg@mamontov.net>

    *   Rusty Conover <https://github.com/rustyconover>

    *   Stefan Hornburg <racke@linuxia.de>

SEE ALSO
    Email::Sender
    MIME::Entity

AUTHORS
    *   Naveed Massjouni <naveed@vt.edu>

    *   Al Newkirk <awncorp@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by awncorp.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
==================================================
Changes from 2004-05-31 00:00:00 +0000 to present.
==================================================

-------------------------------------------
version 1.0201 at 2014-05-29 04:41:50 +0000
-------------------------------------------

  Change: f2d19c7db86bcdc2ce19574bff0103b7e679954e
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-05-29 00:41:50 +0000

    releasing v1.0201 

-------------------------------------------
version 1.0200 at 2014-05-29 04:24:56 +0000
-------------------------------------------

  Change: 7e7916e11bddb89ac799e77bc488566dbbff7308
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-05-29 00:24:56 +0000

    releasing v1.0200 

  Change: d8e23eef1ebceee3a73c1f59743c53f3ebac907b
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-05-17 16:32:50 +0000

    Merge pull request #19 from fany/master

    added possibility to set sender address 

  Change: 593067fd7a7e7b9d66b8214bd76239ac92631c21
  Author: Martin H. Sluka <martin@sluka.de>
  Date : 2014-05-13 16:04:41 +0000

    added possibility to set sender address

    (different from From header) 

  Change: 2f799f8bf4c82257c1336e8c375b33c75443a458
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-04-24 01:08:07 +0000

    Merge pull request #18 from melmothx/feature/multipart

    Added "multipart" key 

  Change: 3e2e75555b67b6512f6db69e34736130314cf38d
  Author: Marco Pessotto <melmothx@gmail.com>
  Date : 2014-04-15 10:15:08 +0000

    Cleaned up and documented 

  Change: 3ad998a6a8cf95149fdef27ae6817a2569241175
  Author: Marco Pessotto <melmothx@gmail.com>
  Date : 2014-04-15 09:41:35 +0000

    Permit the usage of make_multipart('related'); 

-------------------------------------------
version 1.0101 at 2014-02-09 22:38:22 +0000
-------------------------------------------

  Change: bc4429f7fef86b44999f31b177586922d4be7650
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-02-09 17:38:22 +0000

    releasing v1.0101 

  Change: a8a84a5266cec6062cc99b1a57b4d7b0ea1996c3
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2014-02-09 17:05:16 +0000

    Merge pull request #17 from rustyconover/master

    Prevent subroutine prototype mismatch between Dancer's load and
    Module::Load 

  Change: e971922f1d042f4286f05f7a1f829e8a13fe79ef
  Author: Rusty Conover <rusty@luckydinosaur.com>
  Date : 2014-02-09 09:42:36 +0000

    Change to only import debug and warning from Dancer rather than disallowing
    load. 

  Change: 6be5db6d2a8530e7f1aa5145726e8cad8eae80aa
  Author: Rusty Conover <rusty@luckydinosaur.com>
  Date : 2014-02-09 00:09:02 +0000

    Prevent Dancer's load from being exported so that it doesn't conflict with
    Module::Load's load export. 

  Change: 8cbeb4c5a873afe472b8b636146f13ec2a5af957
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2013-10-18 07:40:22 +0000

    Merge pull request #16 from episodeiv/master

    Provide Date header with a sensible default value 

  Change: 24e40fddfb91adc7b61fa3a6eefa96df9d6a0bf5
  Author: Dennis LichtenthaÌler <dennis.lichtenthaeler@episode-iv.de>
  Date : 2013-10-18 13:54:50 +0000

    Provide Date header with a sensible default value 

  Change: 28ae884d8df76bad707b748c82a89dafbabfce03
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2013-03-09 00:00:15 +0000

    Getting ready to release version 1.0100 

  Change: 35cb9ade327c73c7d39ce291875b67c869d7461a
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2013-03-08 20:28:40 +0000

    Merge pull request #14 from melmothx/master

    Make the MIME fields for attachments accessible from the plugin 

  Change: 6d397e90e49a18fe06c1bfa66d8444d3d5767b36
  Author: Marco Pessotto <melmothx@gmail.com>
  Date : 2013-02-27 17:08:51 +0000

    untabified the file 

  Change: ea8e9b2ae95ce3cb7bc17c31d801b25334f0ebc6
  Author: Marco Pessotto <melmothx@gmail.com>
  Date : 2013-02-09 13:15:59 +0000

    Give access to MIME fields to the user for the attachments

    This patch doesn't break retrocompatibility. When passing an hashref to the
    attachment, pass the hash to MIME::Entity, complaining if Path is missing.

    If a string is passed, the original routine is used.

    The example in the documentation has been changed to show a possible use
    (embedding CID images in HTML mails). 

-------------------------------------------
version 1.0000 at 2012-09-16 08:42:11 +0000
-------------------------------------------

  Change: cd6aad8db5f7a73f65ffe7c2628b524db5d76a48
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2012-09-16 08:42:11 +0000

    Updating version for release 1.0000 

  Change: 1e66a34271e4ebdb162a158ff6922621df82b725
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2012-09-16 08:17:37 +0000

    Added support for type => 'html' and extra headers. 

  Change: ab3abc1e4bbcc724d40180e523a3f3ea40490f9d
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2012-08-17 05:17:34 +0000

    minor cleanup 

  Change: 1cad6a9b9136cf72d2456bc76d16b1f2f9ed2d18
  Author: Stefan Hornburg (Racke) <racke@linuxia.de>
  Date : 2012-07-29 13:51:28 +0000

    Add support for redirecting emails in development environments with
    Email::Sender::Transport::Redirect. 

  Change: d5c10a5e09abcd268343bca76566e58d5e06f0f4
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2012-07-22 04:39:13 +0000

    Added support for file attachments 

  Change: 160d760a9fcae1d5220c30c8aba259dc2fd6390b
  Author: Naveed Massjouni <naveedm9@gmail.com>
  Date : 2012-07-16 02:36:32 +0000

    Major refactor which uses Email::Sender instead of Email::Stuff

    Email::Stuff is built on Email::Send which is deprecated in favor of
    Email::Sender. This is a major refactor which basically re-implements the
    entire module on top of Email::Sender. This is a breaking change. I tried
    my best to keep the interface the same (email { ... }), but the

eg/Test/Test.pl
#!C:\strawberry\perl\bin\perl.exe
use Dancer;
use lib ('../../lib');
use lib ('lib');
load_app 'Test';
dance;


eg/Test/Makefile.PL
use strict;
use warnings;
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME                => 'Test',
    AUTHOR              => q{YOUR NAME <youremail@example.com>},
    VERSION_FROM        => 'lib/Test.pm',
    ABSTRACT            => 'YOUR APPLICATION ABSTRACT',
    ($ExtUtils::MakeMaker::VERSION >= 6.3002
      ? ('LICENSE'=> 'perl')
      : ()),
    PL_FILES            => {},
    PREREQ_PM => {
        'Test::More' => 0,
        'YAML'       => 0,
        'Dancer'     => 1.1811,
    },
    dist                => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean               => { FILES => 'Test-*' },
);


eg/Test/lib/Test.pm
package Test;

use Dancer ':syntax';
use Dancer::Plugin::Email;
use Data::Dumper qw/Dumper/;

our $VERSION = '0.1';

get '/' => sub {
    
    die Dumper email {
        to      => 'anewkirk@palcs.org',
        subject => 'Dancer::Plugin::Email test',
        message => 'This is only a test, cool?'
    };
    
};

true;


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.013.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Simple email sending for Dancer applications",
  "AUTHOR" => "Naveed Massjouni <naveed\@vt.edu>, Al Newkirk <awncorp\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dancer-Plugin-Email",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dancer::Plugin::Email",
  "PREREQ_PM" => {
    "Authen::SASL" => "2.16",
    "Dancer" => 0,
    "Dancer::Plugin" => 0,
    "Email::Date::Format" => 0,
    "Email::Sender::Simple" => 0,
    "File::Type" => 0,
    "MIME::Base64" => "3.14",
    "MIME::Entity" => 0,
    "Module::Load" => 0,
    "Net::SMTP::SSL" => "1.01",
    "Net::SSLeay" => "1.58"
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.0201",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Authen::SASL" => "2.16",
  "Dancer" => 0,
  "Dancer::Plugin" => 0,
  "Email::Date::Format" => 0,
  "Email::Sender::Simple" => 0,
  "File::Spec" => 0,
  "File::Type" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "MIME::Base64" => "3.14",
  "MIME::Entity" => 0,
  "Module::Load" => 0,
  "Net::SMTP::SSL" => "1.01",
  "Net::SSLeay" => "1.58",
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);





lib/Dancer/Plugin/Email.pm
package Dancer::Plugin::Email;

our $VERSION = '1.0201'; # VERSION

use Dancer qw(:syntax debug warning);
use Dancer::Plugin;
use Email::Sender::Simple 'sendmail';
use Email::Date::Format 'email_date';
use File::Type;
use MIME::Entity;
use Module::Load 'load';

register email => sub {
    my $params = shift || {};
    my $multipart = delete $params->{multipart};
    my $extra_headers = delete($params->{headers}) || {};
    my $conf = plugin_setting;
    my $conf_headers = $conf->{headers} || {};
    my %headers = ( %$conf_headers, %$params, %$extra_headers );
    my $attach = $headers{attach};
    my $sender = delete $headers{sender};
    if (my $type = $headers{type}) {
        $headers{Type} = $type eq 'html' ? 'text/html' : 'text/plain';
    }
    $headers{Type}   ||= 'text/plain';
    $headers{Format} ||= 'flowed' if $headers{Type} eq 'text/plain';
    $headers{Date}   ||= email_date();
    delete $headers{$_} for qw(body message attach type);

    my $email = MIME::Entity->build(
        Charset  => 'utf-8',
        Encoding => 'quoted-printable',
        %headers, # %headers may overwrite type, charset, and encoding
        Data => $params->{body} || $params->{message},
    );
    if ($attach) {
        if ($multipart) {
            # by default, when you add an attachment,
            # C<make_multipart> will be called by MIME::Entity, but
            # defaults to 'mixed'. Thunderbird doesn't like this for
            # embedded images, so we have a chance to set it to
            # 'related' or anything that the user wants
            $email->make_multipart($multipart);
        }
        my @attachments = ref($attach) eq 'ARRAY' ? @$attach : $attach;
        for my $attachment (@attachments) {
            my %mime;
            if (ref($attachment) eq 'HASH') {
                %mime = %$attachment;
                unless ($mime{Path}) {
                    warning "No Path provided for this attachment!";
                    next;
                };
                $mime{Encoding} ||= 'base64';
                $mime{Type} ||= File::Type->mime_type($mime{Path}),
            } else {
                %mime = (
                    Path     => $attachment,
                    Type     => File::Type->mime_type($attachment),
                    Encoding => 'base64',
                );
            }
            $email->attach(%mime);
        }
    }

    my $transport;
    my $conf_transport = $conf->{transport} || {};
    if (my ($transport_name) = keys %$conf_transport) {
        my $transport_params = $conf_transport->{$transport_name} || {};
        my $transport_class = "Email::Sender::Transport::$transport_name";
        my $transport_redirect = $transport_params->{redirect_address};
        load $transport_class;
        $transport = $transport_class->new($transport_params);

        if ($transport_redirect) {
            $transport_class = 'Email::Sender::Transport::Redirect';
            load $transport_class;
            debug "Redirecting email to $transport_redirect.";
            $transport = $transport_class->new(
                transport        => $transport,
                redirect_address => $transport_redirect
            );
        }
    }
    my %sendmail_arg = ( transport => $transport );
    $sendmail_arg{from} = $sender if defined $sender;
    return sendmail $email, \%sendmail_arg;
};


register_plugin;

# ABSTRACT: Simple email sending for Dancer applications


1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Dancer::Plugin::Email - Simple email sending for Dancer applications

=head1 VERSION

version 1.0201

=head1 SYNOPSIS

    use Dancer;
    use Dancer::Plugin::Email;
    
    post '/contact' => sub {
        email {
            from    => 'bob@foo.com',
            to      => 'sue@foo.com',
            subject => 'allo',
            body    => 'Dear Sue, ...',
            attach  => '/path/to/attachment',
        };
    };

=head1 DESCRIPTION

This plugin tries to make sending emails from L<Dancer> applications as simple
as possible.
It uses L<Email::Sender> under the hood.
In a lot of cases, no configuration is required.
For example, if your app is hosted on a unix-like server with sendmail
installed, calling C<email()> will just do the right thing.

IMPORTANT: Version 1.x of this module is not backwards compatible with the
0.x versions.
This module was originally built on Email::Stuff which was built on
Email::Send which has been deprecated in favor of Email::Sender.
Versions 1.x and on have be refactored to use Email::Sender.
I have tried to keep the interface the same as much as possible.
The main difference is the configuration.
If there are features missing that you were using in older versions,
then please let me know by creating an issue on 
L<github|https://github.com/ironcamel/Dancer-Plugin-Email>.

=head1 FUNCTIONS

This module by default exports the single function C<email>.

=head2 email

This function sends an email.
It takes a single argument, a hashref of parameters.
Default values for the parameters may be provided in the headers section of
the L</CONFIGURATION>.
Paramaters provided to this function will override the corresponding
configuration values if there is any overlap.
An exception is thrown if sending the email fails,
so wrapping calls to C<email> with try/catch is recommended.

    use Dancer;
    use Dancer::Plugin::Email;
    use Try::Tiny;

    post '/contact' => sub {
        try {
            email {
                sender  => 'bounces-here@foo.com', # optional
                from    => 'bob@foo.com',
                to      => 'sue@foo.com, jane@foo.com',
                subject => 'allo',
                body    => 'Dear Sue, ...<img src="cid:blabla">',
                multipart => 'related', # optional, see below
                attach  => [
                    '/path/to/attachment1',
                    '/path/to/attachment2',
                    {
                        Path => "/path/to/attachment3",
                        # Path is required when passing a hashref.
                        # See Mime::Entity for other optional values.
                        Id => "blabla",
                    }
                ],
                type    => 'html', # can be 'html' or 'plain'
                # Optional extra headers
                headers => {
                    "X-Mailer"          => 'This fine Dancer application',
                    "X-Accept-Language" => 'en',
                }
            };
        } catch {
            error "Could not send email: $_";
        };
    };

=head1 CONFIGURATION

No configuration is necessarily required.
L<Email::Sender::Simple> tries to make a good guess about how to send the
message.
It will usually try to use the sendmail program on unix-like systems
and SMTP on Windows.
However, you may explicitly configure a transport in your configuration.
Only one transport may be configured.
For documentation for the parameters of the transport, see the corresponding
Email::Sender::Transport::* module.
For example, the parameters available for the SMTP transport are documented
here L<Email::Sender::Transport::SMTP/ATTRIBUTES>.

You may also provide default headers in the configuration:

    plugins:
      Email:
        # Set default headers (OPTIONAL)
        headers:
          sender: 'bounces-here@foo.com'
          from: 'bob@foo.com'
          subject: 'default subject'
          X-Mailer: 'MyDancer 1.0'
          X-Accept-Language: 'en'
        # Explicity set a transport (OPTIONAL)
        transport:
          Sendmail:
            sendmail: '/usr/sbin/sendmail'

Example configuration for sending mail via Gmail:

    plugins:
      Email:
        transport:
          SMTP:
            ssl: 1
            host: 'smtp.gmail.com'
            port: 465
            sasl_username: 'bob@gmail.com'
            sasl_password: 'secret'

Use the Sendmail transport using the sendmail program in the system path:

    plugins:
      Email:
        transport:
          Sendmail:

Use the Sendmail transport with an explicit path to the sendmail program:

    plugins:
      Email:
        transport:
          Sendmail:
            sendmail: '/usr/sbin/sendmail'

=head2 Multipart messages

You can embed images in HTML messages this way: first, set the C<type>
to C<html>. Then pass the attachments as hashrefs, setting C<Path> and
C<Id>. In the HTML body, refer to the attachment using the C<Id>,
prepending C<cid:> in the C<src> attribute. This works for popular
webmail clients like Gmail and OE, but is not enough for Thunderbird,
which wants a C<multipart/related> mail, not the default
C<multipart/mixed>. You can fix this adding the C<multipart> parameter
set to C<related>, which set the desired subtype when you pass
attachments.

Example:

  email {
         from    => $from,
         to      => $to,
         subject => $subject,
         body    => q{<p>Image embedded: <img src="cid:mycid"/></p>},
         type    => 'html',
         attach  => [ { Id => 'mycid', Path => '/path/to/file' }],
         multipart => 'related'
        };

=head1 CONTRIBUTORS

=over

=item *

Marco Pessotto <melmothx@gmail.com>

=item *

Oleg A. Mamontov <oleg@mamontov.net>

=item *

Rusty Conover <https://github.com/rustyconover>

=item *

Stefan Hornburg <racke@linuxia.de>

=back

=head1 SEE ALSO

=over

=item L<Email::Sender>

=item L<MIME::Entity>

=back

=head1 AUTHORS

=over 4

=item *

Naveed Massjouni <naveed@vt.edu>

=item *

Al Newkirk <awncorp@cpan.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by awncorp.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut
